CIP-0062/README.md

```diff
@@ -0,0 +1,411 @@
+interactions without convenience functions that don't strictly need the wallet's
+state to work.
+
+## api.signVotes(votes: Vote[]): Promise\<Bytes>[]
+
+Errors: [`APIError`](#extended-apierror), [`TxSignError`](#extended-txsignerror)
+
+* votes - an array of up to 10 votes to be validated with the wallet user, and if valid, signed.
+
+IF the wallet user declines SOME of the votes, a [`TxSignError`](#extended-txsignerror) should be raised with `code` set to `VoteRejected` and the optional `rejectedVotes` array specifying the votes rejected.
+
+However, if the wallet user declines the entire set of votes, the wallet should raise a [`TxSignError`](#extended-txsignerror) with the `code` set to `UserDeclined`.
+
+In either case, where the user declines to sign at least 1 of the votes, no signed votes are returned.
+
+### Returns
+
+`Bytes[]` - An array of the hex-encoded strings of the fully encoded and signed vote transactions.  The dApp will submit these votes on behalf of the wallet.
+
+## api.submitDelegation(delegation: Delegation): Promise\<SignedDelegationMetadata>
+
+Errors: [`APIError`](#extended-apierror),
+[`TxSendError`](https://cips.cardano.org/cips/cip30/#txsenderror),
+[`TxSignError`](https://cips.cardano.org/cips/cip30/#txsignerror)
+
+This endpoint should construct the cbor encoded delegation certificate according to the specs in [CIP-36 Example](https://github.com/Zeegomo/CIPs/blob/472181b9c69feeedae0b5b2db8b42d0cf4eb1a11/CIP-0036/README.md#example).
+
+It should then sign the certificate with the staking key as described in the same example as above.
+
+The implementation of this endpoint can make use of the already existing [CIP-30](https://cips.cardano.org/cips/cip30/) `api.signData` and `api.submitTx` to perform the broadcasting of the transaction containing the metadata.
+
+Upon submission of the transaction containing the delegation cert as part of metadata, the wallet should store the delegation in its local storage and return an object that contains the delegation cert, the signature and the txhash of the transaction that the certificate was submitted with.
+
+* delegation - the voter registration [delegation](#delegation) record.
+
+This should be a call that implicitly cbor encodes the `delegation` object and uses the already existing [CIP-30](https://cips.cardano.org/cips/cip30/) `api.submitTx` to submit the transaction. The resulting transaction hash should be returned.
+
+This should trigger a request to the user of the wallet to approve the transaction.
+
+### Returns
+
+The [Signed Delegation Metadata](#signeddelegationmetadata) of the voter registration delegation passed in the `delegation` parameter.
+
+# Examples of Message Flows and Processes
+
+## Delegation Cert process
+
+1. **`Get Voting Key`** - dApp call the method [**api.getCurrentVotingKey**](#apigetcurrentvotingkey-promisecborpublickey) to return a ed25519 32 bytes public key (x value of the point on the curve).
```

Comment:
I can't find the `api.getCurrentVotingKey` method in this doc anymore, was it dropped accidentally?

Reply:
Oops. Correct. fixing